### PR TITLE
Container Topology test + Framework updates

### PR DESCRIPTION
**Pull request by gshefer**
Tests for containers Topology page + Framework updates
{{pytest: cfme/tests/containers/test_topology.py  -v --use-provider container }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 415240860b6ab2370a94fed1e9d92e08800e1454:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f46fbab5778e34bfac38dae8ae167fb3394a6455:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 83d56737ddccb558ed3bcafcd110fcd7c7c6e5ba:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9ac941831406ae788c93835e4b2056ccb78da849:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/web_ui/topology.py [hunk 1]
```diff
@@ -112,11 +115,15 @@ def set_active(self, active=True):
 
 
 class TopologyDisplayNames(object):
-    DISPLAY_NAME = \
-        "//*[contains(@class, 'container_topology')]//label[contains(., 'Display Names')]/input"
+    DISPLAY_NAME_XPATHES = \
```
Comment (gshefer) [inline cfme/web_ui/topology.py]: Done.

**Comment by dajoRH** [comment]
Lint report for commit d58346fdf2671e612fb8a56389ca61c2c63d9dfc:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit eb5f7dbc96b355e26f8b309b422dd3c102c15e1b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3aa80a08a724d699402d75890e05911b734e750c:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bf276c4c6b165e30e66698eb95d4fcbd46b83ffb:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 985df55168a30a655e6ca6a99cca4b3b8d586cb2:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/containers/topology.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from cfme.common import TopologyMixin
+from cfme.fixtures import pytest_selenium as sel
+
+
+class ContainerTopology(TopologyMixin):
```
Comment (mshriver) [inline cfme/containers/topology.py]: @gshefer @psav I just created a new PR for this class. 

Included this classmethod, but replaced the navigation with a proper navmazing destination.

https://github.com/ManageIQ/integration_tests/pull/3758

File: cfme/tests/containers/test_topology.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+def test_topology_display_names():
+    '''Testing Display Names functionality in Topology view
+    Step:
+        In Topology screen click on the Display names check box
+    Expected result:
+            The entities names should toggle on/off
+            according to the state of the checkbox.'''
+    topo_obj = Topology(ContainerTopology)
+
+    topo_obj.display_names.enable(True)
```
Comment (mshriver) [inline cfme/tests/containers/test_topology.py]: consider paramaterizing on True/False and asserting is_displayed_text against the parameter. Reduces the code duplication here.

File: cfme/web_ui/topology.py [hunk 4]
```diff
@@ -112,11 +115,15 @@ def set_active(self, active=True):
...
-        "//*[contains(@class, 'container_topology')]//label[contains(., 'Display Names')]/input"
+    DISPLAY_NAME_XPATHS = \
+        ["//*[contains(@class, 'container_topology')]//label[contains(., 'Display Names')]/input",
+         '//*[@id="box_display_names"]']  # [0] is not working on containers topology
 
     def __init__(self):
-        self._el = sel.element(self.DISPLAY_NAME)
+        try:
+            self._el = sel.element(self.DISPLAY_NAME_XPATHS[0])
+        except:
```
Comment (mshriver) [inline cfme/web_ui/topology.py]: Please narrow the except to the type you're anticipating being thrown here.

File: cfme/tests/containers/test_topology.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        entities within the Topology map
+        should be hidden/shown as per selection'''
+    topo_obj = Topology(ContainerTopology)
+    for legend_name in topo_obj.legends:
+        legend = topo_obj.__getattribute__(legend_name)
+        legend.set_active(False)
+        topo_obj.reload_elements()
+        for elem in topo_obj.elements():
+            # [:-1] because the 's' in the end, which is redundant
+            if elem.type == legend.name[:-1]:
```
Comment (mshriver) [inline cfme/tests/containers/test_topology.py]: consider using rstrip so you don't inadvertently remove a non-s.

**Comment by dajoRH** [comment]
Lint report for commit 0f705735ac3eb1c3ca459e7ea6a708af72cc4a4f:

`cfme/tests/containers/test_topology.py`:
- :heavy_exclamation_mark: Line 28:1: **W293** *blank line contains whitespace*
- :heavy_exclamation_mark: Line 79:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 567a11064b3d4219bdc1c1caad7d37bf49b0d3b4:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/containers/test_topology.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+def test_topology_toggle_display():
+    '''Testing display functionality in Topology view
+    Step:
+        For each legend click "enable/disable"
+    Expected result:
+        entities within the Topology map
+        should be hidden/shown as per selection'''
+    topo_obj = Topology(ContainerTopology)
+    for legend_name in topo_obj.legends:
+        legend = topo_obj.__getattribute__(legend_name)
```
Comment (gshefer) [inline cfme/tests/containers/test_topology.py]: what is the functional difference?

File: cfme/containers/topology.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+
+    @classmethod
+    def load_topology_page(cls):
+        sel.force_navigate('containers_topology')
```
Comment (gshefer) [inline cfme/containers/topology.py]: no

**Comment by dajoRH** [comment]
Lint report for commit 2f9d24362f9fd7fa8c20f73d94b25cf555e02b5e:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ad62189725e9f3b87ec3d4600973cb628c9cdfb0:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 837bd19648ccb49985e2e694b8b3f9d5b0865d1e:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fddf4d4f28ddceb33b3eeb8e1b091f39376ea870:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5a030988a6d275c47af537f3fbbe0f8b6836d173:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/web_ui/topology.py [hunk 8]
```diff
@@ -131,8 +138,8 @@ def disable(self):
 
 
 class TopologySearchBox(object):
-    SEARCH_BOX = "//*[contains(@class, 'container_topology')]//input[@id='search']"
-    SEARCH_CLEAR = "//*[contains(@class, 'container_topology')]//button[contains(@class, 'clear')]"
+    SEARCH_BOX = "//input[@id='search_topology']"
```
Comment (gshefer) [inline cfme/web_ui/topology.py]: this is the ID AFAIK right?

**Comment by dajoRH** [comment]
Lint report for commit 4e8a604e4751d284a4d0a5ab50ef3ba7e2b147d0:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/containers/test_topology.py [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+        for bool_ in (True, False):
+            legend.set_active(bool_)
+            topo_obj.reload_elements()
+            for elem in topo_obj.elements():
+                # [:-1] because the 's' in the end, which is redundant
```
Comment (gshefer) [inline cfme/tests/containers/test_topology.py]: Done :)

**Comment by dajoRH** [comment]
Lint report for commit 3b61cff139da92d72e19c19d09458e064706abde:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0bbfa364a0ce3d4ce8f339e16cc3ab92baddb8e7:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7e6865259b02a69de3d6e0f227536a680ed4c8e5:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit becd1764c925180cde905a98ca161cd0f6e4fb30:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ef9a13049c8026f63fae86b2f94c6e73ba2749c3:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 33243e3cecd3a4d664bc3d0c5f110205a74f4f9e:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cf085320be4f6889edac787369298ba14c0d45c1:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request merged**